RollGen/DomainDice.cs

```diff
@@ -10,17 +10,13 @@ namespace RollGen
     internal class DomainDice : Dice
     {
         private PartialRollFactory partialRollFactory;
-        private Regex expressionRegex;
-        private Regex strictRollRegex;
-        private Regex lenientRollRegex;
+        private readonly Regex expressionRegex = new Regex(RegexConstants.ExpressionWithoutDieRollsPattern);
```

Comment:
These are the ones that I do not like, as instead of being in the constructor, now these assignments happen at the variable declarations.  The `readonly` is absolutely fine, but I would prefer the `new` statements live in the constructor.